internal/util/vault.go

```diff
@@ -268,6 +292,25 @@ func (vc *vaultConnection) Destroy() {
 	}
 }
 
+// getDeleteKeyContext creates a new KeyContext that has an optional value set
+// to destroy the contents of secrets. This is configurable with the
+// `vaultDestroyKeys` configuration parameter.
+//
+// Setting the option `DestroySecret` option in the KeyContext while creating
+// new keys, causes failures, so the option only needs to be set in the
+// RemoveDEK() calls.
+func (vc *vaultConnection) getDeleteKeyContext() map[string]string {
+	keyContext := map[string]string{}
+	for k, v := range vc.keyContext {
+		keyContext[k] = v
+	}
```

Comment:
@nixpanic do we need this new map generation or add key to existing map?

Reply:
Adding it to the existing map will cause failures during non-delete operations. So, creating a copy of the map and adding it there.